cryptolib/attestation.go

```diff
@@ -0,0 +1,89 @@
+/*
+Copyright 2020 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cryptolib
+
+type (
+	// KeyMode stores the type of the Attestation.
+	KeyMode int
+	// SigningAlgorithm stores the public-key signing algorithm and hashing
+	// function used to sign the Attestation.
+	SigningAlgorithm int
+)
+
+// Enumeration of KeyMode
+const (
+	UnknownKeyMode KeyMode = iota
+	Pgp
+	Pkix
+	Jwt
+)
+
+// Enumeration of SigningAlgorithm
+const (
```

Comment:
Are the signing algorithms only for PKIX? If so, should we move them to a PKIX file?

Reply:
+1

Reply:
We opted to put all the algorithms together in a single `signature_algorithm.go` file. This should make it easy to add/remove new algorithms without digging into other files to adjust enum values.